sql/core/src/test/scala/org/apache/spark/sql/sources/TableScanSuite.scala

```diff
@@ -345,34 +345,72 @@ class TableScanSuite extends DataSourceTest with SharedSQLContext {
       (1 to 10).map(Row(_)).toSeq)
   }
 
+  test("create a temp table that does not have a path in the option") {
+    Seq("TEMPORARY VIEW", "TABLE").foreach { tableType =>
+      val tableName = "relationProvierWithSchema"
+      withTable(tableName) {
+        sql(
+          s"""
+             |CREATE $tableType $tableName
```

Comment:
My original thought is to provide a comprehensive test coverage of data source table creation/insertion with `path`. This is not related to this PR. Let me get rid of it.